msal/tests/Test.MSAL.NET.Unit/UnifiedCacheTests.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            private byte[] data;
+            public byte[] LoadCache()
+            {
+                return data;
+            }
+
+            public void WriteCache(byte[] serializedCache)
+            {
+                data = serializedCache;
+            }
+        }
+
+        [TestMethod]
+        [Description("Test unified token cache")]
+        public void UnifedCache_MsalStoresToAndReadRtFromAdalCache()
+        {
+            PublicClientApplication app = new PublicClientApplication(TestConstants.ClientId);
+
+            app.UserTokenCache.legacyCachePersistance = new TestLegacyCachePersistance();
+
+            MockWebUI ui = new MockWebUI()
+            {
+                MockResult = new AuthorizationResult(AuthorizationStatus.Success,
+                    TestConstants.AuthorityHomeTenant + "?code=some-code")
+            };
+
+            MsalMockHelpers.ConfigureMockWebUI(new AuthorizationResult(AuthorizationStatus.Success,
+                app.RedirectUri + "?code=some-code"));
+
+            //add mock response for tenant endpoint discovery
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityHomeTenant)
+            });
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Post,
+                ResponseMessage = MockHelpers.CreateSuccessTokenResponseMessage()
+            });
+
+            AuthenticationResult result = app.AcquireTokenAsync(TestConstants.Scope).Result;
+            Assert.IsNotNull(result);
+
+            // make sure Msal stored RT in Adal cache
+            IDictionary<AdalTokenCacheKey, AdalResultWrapper> adalCacheDictionary = 
+                AdalCacheOperations.Deserialize(app.UserTokenCache.legacyCachePersistance.LoadCache());
+
+            Assert.IsTrue(adalCacheDictionary.Count == 1);
+
+            // clear Msal cache
+            app.UserTokenCache.ClearCache();
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Post,
+                PostData = new Dictionary<string, string>()
+                {
+                    {"grant_type", "refresh_token"}
+                },
+                ResponseMessage =
+                    MockHelpers.CreateSuccessTokenResponseMessage(TestConstants.UniqueId,
+                        TestConstants.DisplayableId,
+                        TestConstants.Scope.ToArray())
+            });
+
+            // Using RT from Adal cache for silent call
+            AuthenticationResult result1 = app.AcquireTokenSilentAsync
+                (TestConstants.Scope, result.User, TestConstants.AuthorityCommonTenant, false).Result;
+
+            Assert.IsNotNull(result1);
```

Comment:
don't you need to validate that the result returned is in more details what is expected? (btw. same comment above)